package.json

```diff
@@ -34,189 +33,9 @@
-		"escape-regexp": "0.0.1",
-		"escape-string-regexp": "2.0.0",
-		"events": "3.0.0",
-		"exports-loader": "0.7.0",
-		"express": "4.17.1",
-		"express-useragent": "1.0.13",
-		"fast-json-stable-stringify": "2.0.0",
-		"filesize": "6.0.1",
-		"flag-icon-css": "3.4.5",
-		"flux": "3.1.3",
-		"fontfaceobserver": "2.1.0",
-		"fuse.js": "3.4.6",
-		"get-video-id": "3.1.4",
-		"gfm-code-blocks": "1.0.0",
-		"globby": "10.0.1",
-		"gridicons": "3.3.1",
-		"gzip-size": "5.1.1",
-		"hash.js": "1.1.7",
-		"he": "1.2.0",
-		"html-loader": "0.5.5",
-		"html-to-react": "1.4.2",
-		"html-webpack-plugin": "3.2.0",
-		"i18n-calypso": "file:./packages/i18n-calypso",
-		"ignore-loader": "0.1.2",
-		"immutability-helper": "3.0.1",
-		"immutable": "3.8.2",
-		"imports-loader": "0.8.0",
-		"inherits": "2.0.4",
-		"is-my-json-valid": "2.20.0",
-		"isomorphic-fetch": "2.2.1",
-		"jquery": "1.12.3",
-		"keymaster": "1.6.2",
-		"lerna": "3.18.4",
-		"loader-utils": "1.2.3",
-		"lodash": "4.17.15",
-		"lodash-es": "4.17.15",
-		"lru": "3.1.0",
-		"lunr": "2.3.8",
-		"marked": "0.7.0",
-		"mkdirp": "0.5.1",
-		"moment": "2.24.0",
-		"moment-timezone": "0.5.27",
-		"morgan": "1.9.1",
-		"node-sass-package-importer": "5.3.2",
-		"npm-run-all": "4.1.5",
-		"objectpath": "1.2.2",
-		"page": "1.11.5",
-		"path-browserify": "1.0.0",
-		"percentage-regex": "3.0.0",
-		"phone": "2.4.2",
-		"photon": "file:./packages/photon",
-		"postcss-cli": "6.1.3",
-		"prismjs": "1.17.1",
-		"prop-types": "15.7.2",
-		"qrcode.react": "1.0.0",
-		"qs": "6.9.1",
-		"react": "16.12.0",
-		"react-click-outside": "3.0.1",
-		"react-day-picker": "7.4.0",
-		"react-dom": "16.12.0",
-		"react-element-to-jsx-string": "14.1.0",
-		"react-lazily-render": "1.2.0",
-		"react-live": "1.12.0",
-		"react-modal": "3.11.1",
-		"react-redux": "7.1.3",
-		"react-stripe-elements": "4.0.2",
-		"react-transition-group": "4.3.0",
-		"react-virtualized": "9.21.2",
-		"redux": "4.0.4",
-		"redux-form": "8.2.6",
-		"redux-thunk": "2.3.0",
-		"refx": "3.1.1",
-		"rememo": "3.0.0",
-		"resize-observer-polyfill": "1.5.1",
-		"rtlcss": "2.4.0",
-		"social-logos": "2.1.0",
-		"socket.io-client": "2.3.0",
-		"source-map": "0.7.3",
-		"source-map-support": "0.5.16",
-		"store": "2.0.12",
-		"striptags": "3.1.1",
-		"superagent": "3.8.3",
-		"svg4everybody": "2.1.9",
-		"textarea-caret": "3.1.0",
-		"tinymce": "4.9.6",
-		"to-title-case": "1.0.0",
-		"tracekit": "0.4.5",
-		"twemoji": "12.1.4",
-		"url": "0.11.0",
-		"use-debounce": "3.1.0",
-		"use-subscription": "1.3.0",
-		"utility-types": "3.10.0",
-		"uuid": "3.3.3",
-		"valid-url": "1.0.9",
-		"webpack-bundle-analyzer": "3.6.0",
-		"webpack-dev-middleware": "3.7.2",
-		"wpcom": "5.4.2",
-		"wpcom-proxy-request": "5.0.2",
-		"wpcom-xhr-request": "1.1.3",
-		"yamlparser": "0.0.2",
-		"yargs": "13.3.0"
+		"@automattic/full-site-editing": "file:apps/full-site-editing",
+		"@automattic/wpcom-block-editor": "file:apps/wpcom-block-editor",
+		"wp-calypso": "file:client"
```

Comment:
if we go this route we should probably include everything in apps/*?

Reply:
Yes, and also all the packages in `packages/` that are not used by Calypso or something else in the repo transitively.

In other words, all packages in the monorepo should be direct or indirect dependencies of the root `package.json`. That ensures that `npm install` installs the whole package tree for the whole monorepo.

Gutenberg does it: https://github.com/WordPress/gutenberg/blob/master/package.json#L20-L66

Alternatively, `lerna bootstrap` does the npm installation process, but bails out to a simple `npm install` if it sees a local `file:` ref in `dependencies` in the root `package.json`. We could also go that way, but leaving the npm tree solely to the npm client is preferrable.

Reply:
> Alternatively, lerna bootstrap does the npm installation process, but bails out to a simple npm install if it sees a local file: ref in dependencies in the root package.json. We could also go that way, but leaving the npm tree solely to the npm client is preferrable.

Yeah, if we want to use lerna bootstrap, we'd also want to jump back to semver ranges instead of file refs. Once you add one file ref bootstrap stops making sense. If we really want a mix, something like yarn workspaces or pnpm workspaces offer a lot more flexibility.

Reply:
The way I've been thinking about it: lerna provides the "I want to run a command against the package graph" functionality and something else (npm, yarn, pnpm) provides the install and graph linkage. `lerna bootstrap` is to be avoided as it's just a hacky way of doing what one of the package managers would rather do itself.

Reply:
Totally agreed that `lerna bootstrap` is to be avoided. Even the Lerna maintainer told us that it's a buggy and hard-to-maintain hack that tries to do the package manager's job (poorly).